### PR TITLE
cargo-outdated 0.15.0

### PR DESCRIPTION
**Pull request by cho-m**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

New release needed for #166949

### CODE REVIEW

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
- relates to https://github.com/kbknapp/cargo-outdated/issues/388

File: Formula/c/cargo-outdated.rb [hunk 1]
```diff
@@ -1,37 +1,36 @@
...
-  #    - Migrate to the corresponding `libgit2` formula.
-  #    - Change the `LIBGIT2_SYS_USE_PKG_CONFIG` env var below to `LIBGIT2_NO_VENDOR`.
-  #      See: https://github.com/rust-lang/git2-rs/commit/59a81cac9ada22b5ea6ca2841f5bd1229f1dd659.
-  depends_on "libgit2@1.6"
+  depends_on "libgit2"
   depends_on "openssl@3"
 
   def install
-    ENV["LIBGIT2_SYS_USE_PKG_CONFIG"] = "1"
+    system "tar", "--strip-components", "1", "-xzvf", "cargo-outdated-#{version}.crate" if build.stable?
```
Comment (carlocab) [inline Formula/c/cargo-outdated.rb:31]: We should just add crate handling to `brew` I think

Comment (cho-m) [reply Formula/c/cargo-outdated.rb:31]: We could create a subclass of `Tar` and just let `Tar` handle all the logic. I don't think there is an official spec of .crate, but currently it looks like a renamed .tar.gz file.

Comment (cho-m) [reply Formula/c/cargo-outdated.rb:31]: Actually will need a little more work as default `Tar` logic forwards to `Gzip` (running `gunzip`) which doesn't like arbitrary extensions

**Pull request merged**